### DISCUSSION TITLE
How to retrive buyer data from order

### DISCUSSION THREAD

**Discussion by simonebosio91** [abuzuhri/Amazon-SP-API-CSharp#103] [Q&A]
Hi,
first of all congrats for this incredible library.
I've just trying to use the order retriving process, how can I get the complete information of the buyer?
I need **buyer name, surname and address**, but in the amazon response this data are currently not reacheble.
Regards

**Comment by abuzuhri** [comment]
Can you look here

https://github.com/abuzuhri/Amazon-SP-API-CSharp/issues/74

**Comment by simonebosio91** [comment]
Hi I tried to follow the solution in problem 74 but it doesn't work.
I have configured the permissions in amazon ( attached screen),  created new token and then run the call as in problem 74 but I get the error that I cannot access the buyerInfo and shippingAddress resources.
I need to contact amazon to get these permissions?

![image](https://user-images.githubusercontent.com/36471454/154795623-f17e8c31-c3a8-49f7-ac88-4c6e890f9439.png)

**Reply by abuzuhri** [reply]
Can you post the code you used here ...

**Comment by simonebosio91** [comment]
this is the code:
` var parameterOrderList = new ParameterOrderList
            {
                CreatedAfter = DateTime.UtcNow.AddHours(-24),
                OrderStatuses = new List<Constants.OrderStatuses> { Constants.OrderStatuses.Unshipped },
                MarketplaceIds = new List<string> { "APJ6JRA9NG5V4" },
                IsNeedRestrictedDataToken = true,
                RestrictedDataTokenRequest = new CreateRestrictedDataTokenRequest
                {
                    restrictedResources = new List<RestrictedResource>
                    {
                        new RestrictedResource
                        {
                            method = Method.GET.ToString(),
                            path = ApiUrls.OrdersApiUrls.Orders,
                            dataElements = new List<string> { "buyerInfo", "shippingAddress" }
                        }
                    }
                }
            };

            var orders = amazonConnection.Orders.GetOrders(parameterOrderList);`

And this is the error:
![image](https://user-images.githubusercontent.com/36471454/154795790-7bc13e53-1e7d-4050-a6c2-67f864bd3748.png)

**Reply by abuzuhri** [reply]
Please email me

**Reply by haicz** [reply]
Hello,
How is problem resolved?
I have some problem.

**Reply by dafoink** [reply]
I am getting this same problem.

**Reply by abuzuhri** [reply]
you need to have PII approved by amazon first
our library work perfect only you need to get approved by amazon first

**Reply by asifiq2000** [reply]
Hi @abuzuhri  I am also receiving the same error, but when I create the support ticket on amazon 
they asked to provide **full request and full response not truncated attached as a .txt document. 
Along the files, please make sure which API you are using and which operation is being called.**

However I am posting through Fika Api how could I provide the asked details, please advise.

And one more query If amazon approves the PII permission even I got the same error then 
could it be mission permission from my end, if yes then where should I check at my end for
those permissions? I mean should I need to check in my developer App's rights or IAM User's
RoleARN permissions, please suggest what exactly I must check to confirm the permission 
issue is not from my end.
